extensions/amp-story-interactive/0.1/utils.js

```diff
@@ -47,3 +47,38 @@ export const deduplicateInteractiveIds = (doc) => {
     }
   }
 };
+
+/**
+ * Reorders options data to account for scrambled or incomplete data.
+ *
+ * @param {!Array<!InteractiveOptionType>} optionsData
+ * @param {!Array<!Element>} optionElements
+ * @return {!Array<!InteractiveOptionType>}
+ */
+export const orderData = (optionsData, optionElements) => {
+  const orderedData = new Array(optionElements.length);
+  for (let i = 0; i < optionsData.length; i++) {
```

Comment:
In some cases backends can send more options than the number of optionElements, we need to check that `optionsData[i].index` is lower than `optionElements.length`